doc/developer-notes.md

```diff
@@ -834,8 +836,25 @@ GUI
   - *Rationale*: Blocking the GUI thread can increase latency, and lead to
     hangs and deadlocks.
 
-Subtrees
-----------
+Git Usage
+---------
+
+### Rebasing
+
+When the master development branch develops conflicts with your pull request,
+you need to rebase it. This means rebuilding your branch on top of a new base.
+Git has a built-in `rebase` command which can automate much of this.
+
+Sometimes conflicts are invisible to git: the branches will merge cleanly, but
+builds will fail, or functionality will break. These are still considered
+conflicts, and you should rebase when fixing them.
+
+Note that when there are no conflicts at all, however, rebasing should be
+avoided, as it can be disruptive to reviewers. (The `git rebase` command may
```

Comment:
It is no more disruptive than any other force push. `git range-diff` can handle rebases easily